sbi/neural_nets/estimators/flowmatching_estimator.py

```diff
@@ -351,3 +354,427 @@ def std_fn(self, times: Tensor) -> Tensor:
+            estimated by the neural
+        network (see Equation 1 in [1]_ with added conditioning on :math:`x_o`. The
+        notation is changed to match the standard SBI notation:
+            :math:`\text{inputs}_0 = x_0` and :math:`\text{inputs}_1 = x_1`).
+
+        The loss is computed as the mean squared error between the vector field:
+
+        .. math::
+            L(\text{inputs}_0, \text{inputs}_1, t, x_o) = \| v(\text{inputs}_t, t; x_o)
+                - (\text{inputs}_1 - \text{inputs}_0) \|^2
+
+            where
+
+            .. math::
+                \text{inputs}_1 \sim p_{base}
+                \text{inputs}_t = t \cdot \text{inputs}_1 + (1 - t) \cdot
+                    \text{inputs}_0
+
+            Additionally, the small noise :math:`\sigma_{min}` is added to the
+            vector field as per [2]_ to address numerical issues at small times.
+
+        Args:
+            input: Original data
+                Shape: [B, T, F]
+            times: SDE time variable in [t_min, t_max]. Uniformly sampled if None.
+            condition_mask: A boolean mask indicating the role of each variable.
+                Expected shape: `(batch_size, num_variables)`.
+                - `True` (or `1`): The variable at this position is observed and its
+                    features will be used for conditioning.
+                - `False` (or `0`): The variable at this position is latent and its
+                    features are subject to inference.
+            edge_mask: A boolean mask defining the adjacency matrix of the directed
+                acyclic graph (DAG) representing dependencies among variables.
+                Expected shape: `(batch_size, num_variables, num_variables)`.
+                - `True` (or `1`): An edge exists from the row variable to the column
+                    variable.
+                - `False` (or `0`): No edge exists between these variables.
+                - if None, it will be equivalent to a full attention (i.e., full ones)
+                    mask, we suggest you to use None instead of ones
+                    to save memory resources
+            rebalance_loss: If True, the loss for each batch item is divided by the
+                number of latent (unobserved) variables in that item. This is useful
+                when the number of latent variables varies across the batch, as it
+                prevents items with more latent variables from dominating the loss.
+
+        Returns:
+            Loss value.
+        """
+        # randomly sample the time steps to compare the vector field at
+        # different time steps
+        if times is None:
+            times = torch.rand(input.shape[:-2], device=input.device, dtype=input.dtype)
+        times_ = times[..., None, None]
+
+        # sample from probability path at time t
+        input_1 = torch.randn_like(input)
+        input_t = (1 - times_) * input + (times_ + self.noise_scale) * input_1
```

Comment:
Just form our previous discussion. The bug in flow matching is here: We cannot add noise here on things we condition (i.e. where the condition mask is true.

Reply:
So you also have to keep the input the same i.e. `input_t = torch.where(conditon_mask, input, input_t)`